source/common/tracing/zipkin/span_buffer.cc

```diff
@@ -0,0 +1,30 @@
+#include "common/tracing/zipkin/span_buffer.h"
+
+namespace Zipkin {
+
+bool SpanBuffer::addSpan(SpanPtr&& span) {
+  if (span_buffer_.size() == span_buffer_.capacity()) {
+    // Buffer full
+    return false;
+  }
+  span_buffer_.push_back(std::move(span));
+
+  return true;
+}
+
+std::string SpanBuffer::toStringifiedJsonArray() {
+  std::string stringified_json_array = "[";
+
+  if (pendingSpans()) {
+    stringified_json_array += span_buffer_[0]->toJson();
+    const uint64_t size = span_buffer_.size();
+    for (uint64_t i = 1; i < size; i++) {
+      stringified_json_array += ",";
+      stringified_json_array += span_buffer_[i]->toJson();
+    }
```

Comment:
Well, Json::Factory::listAsJsonString() does not do exactly the same thing. As a result of that method, the elements of the stringified json array are stringified as well, which means that an extra-level of double-quote escaping would be present. That was not my intention.
I would, therefore, prefer to keep this code unchanged.